src/lib/storage/storage.ts

```diff
@@ -289,6 +298,31 @@ export interface CredentialMetadataSummary extends CredentialMetadata {
   issuer: IdentitySummary
 }
 
+const cacheDIDDoc = (connection: Connection) => (doc: DidDocument) => {
+  const cacheEntry = plainToClass(CacheEntity, {
+    // TODO Change to did?
+    key: `didCache:${doc.did}`,
```

Comment:
So a higher order function, e.g. `withCacing` which takes a function to generate the key / do the storing? 😄

Reply:
we could solidify it with a generic `cacheType(type: string) => (key: string, value: {})` where type just gets prepended to the key and you could generate new cache functions a la `const cacheDidDoc = cacheType('didDoc')`, this would be quick and fairly lightweight. you could also make it generic over the value type with `cacheType<T>(type: string) => (key: string, value: T)`

Reply:
Yeah not sure if the decorator-style semantics make sense here, mostly because I'm not sure what would get decorated!

Something like @chunningham's proposal makes more sense IMHO so far.

Reply:
This is probably off-topic, moved it here: #1420